packages/curve-ui-kit/src/hooks/useIsMobile.ts

```diff
@@ -0,0 +1,96 @@
+import { useEffect, useState } from 'react'
+
+/** Detects if the current device is a mobile device based on touch capabilities */
+function detectIsMobile() {
+  let hasTouchScreen
+
+  if ('maxTouchPoints' in navigator) {
+    hasTouchScreen = navigator.maxTouchPoints > 0
+  } else if ('msMaxTouchPoints' in navigator) {
+    // @ts-ignore
+    hasTouchScreen = navigator.msMaxTouchPoints > 0
+  } else {
+    const mQ = matchMedia?.('(pointer:coarse)')
+    if (mQ?.media === '(pointer:coarse)') {
+      // @ts-ignore
+      hasTouchScreen = mQ.matches
+    } else if ('orientation' in window) {
+      hasTouchScreen = true // deprecated, but good fallback
+    } else {
+      // Only as a last resort, fall back to user agent sniffing
+      // @ts-ignore
+      const UA = navigator.userAgent
+      hasTouchScreen =
+        /\b(BlackBerry|webOS|iPhone|IEMobile)\b/i.test(UA) || /\b(Android|Windows Phone|iPad|iPod)\b/i.test(UA)
+    }
+  }
+
+  return hasTouchScreen
+}
+
+// Singleton state management
+const subscribers = new Set<(value: boolean) => void>()
+let isMobileValue = typeof window !== 'undefined' ? detectIsMobile() : false
+let isListenerAttached = false
+
+/** Updates the mobile state and notifies all subscribers */
+function setIsMobile(value: boolean): void {
+  if (value !== isMobileValue) {
+    isMobileValue = value
+    subscribers.forEach((callback) => callback(value))
+  }
+}
+
+// Initialize global listener
+if (typeof window !== 'undefined' && !isListenerAttached) {
+  window.addEventListener('resize', () => {
```

Comment:
This does not make sense if we are checking `maxTouchPoints` - that won't change regardless of the screen size.
If we want to measure the screen size we should use mui's `useMediaQuery`

Reply:
Completely got rid of the custom `useIsMobile` hook and replaced all with const `useMediaQuery((t) => t.breakpoints.down('tablet'))`